internal/elasticsearch/ingest/processors.go

```diff
@@ -0,0 +1,245 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package ingest
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"sort"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// Processor represents an ingest processor.
+type Processor struct {
+	// Type of processor ("set", "script", etc.)
+	Type string `yaml:"-"`
+	// FirstLine is the line number where this processor definition starts
+	// in the pipeline source code.
+	FirstLine int `yaml:"-"`
+	// LastLine is the line number where this processor definitions end
+	// in the pipeline source code.
+	LastLine int `yaml:"-"`
+}
+
+// Processors return the list of processors in an ingest pipeline.
+func (p Pipeline) Processors() (procs []Processor, err error) {
+	switch p.Format {
+	case "yaml", "yml":
+		procs, err = processorsFromYAML(p.Content)
+	case "json":
+		procs, err = processorsFromJSON(p.Content)
```

Comment:
I see that YAML processing has much simpler logic than JSON processing. Do both formats differ much from each other in terms of properties? JSON is a subset of YAML. If there are differences, maybe it's valid to drop a comment here, justifying why there are two separate functions.

Reply:
It's much simpler now, tons of code removed, good job!